Refactor "tx too old" tests

nickva opened the pull request:
* There was a good amount of duplication between `_db_crud_tests` and
 `_changes_fold_tests`, so make a common test utility module so both suites can
 use.

* Clean up test names. Previously some were named `tx_too_long` but since the
  official FDB error is `transaction_too_old` rename them to match a bit
  better.

* `list_dbs_info` implementation uses queue of 100 futures to parallelize fetching.
  So its test was updated to create more than 100 dbs. Creating 100 dbs took
  about 3 seconds so added a small parallel map (pmap) utility function to help
  with that.

To test:

   `make check-fdb`

jaydoane commented on src/fabric/test/fabric2_tutil.erl:
```diff
@@ -0,0 +1,99 @@
+% Licensed under the Apache License, Version 2.0 (the "License"); you may not
```
At the risk of bike shedding, `tutil` seems a little terse, and I'd prefer `fabric2_test_util` if it were up for a vote.

nickva replied:
Changed to fabric2_test_util

nickva commented:
Added a minor fixup. @davisp suggested to enhance pmap to make it properly re-throw exceptions so it behaves closer to `lists:map/2. I had forgotten to update the original PR with that.

Also I moved it to `fabric2_util` so it can more generally useful.

As a benefit I could remove the extra "validate pmap results" steps from the tests.
